src/value.rs

```diff
@@ -272,9 +261,14 @@ pub(crate) fn string_parts(string: &types::Str) -> Vec<StrPart> {
         if let StrPart::Literal(ref mut text) = part {
             if multiline {
                 *text = remove_indent(text, i == 0, common);
+                // If the last literal is of the form `X\nY`, exclude Y if it consists solely of spaces
                 if i == literals - 1 {
-                    // Last index
-                    remove_trailing(text);
+                    match text.rfind('\n') {
+                        Some(p) if text[p + 1..].chars().all(|c| c == ' ') => {
```

Comment:
This does not recognize tabs and such as whitespace, I assume this is intentional.

Reply:
Right, really I should say "spaces" and not "whitespace." If the last line has some other whitespace, like a tab character, it should not be removed. This matches the behavior of the reference implementation:
```
# test.nix
''
hello
	'' # this is a tab
```
```
$ nix-instantiate --parse test.nix
"hello\n\t"
```